main/src/main/java/sbt/internal/ReverseLookupClassLoader.java

```diff
@@ -64,6 +66,31 @@ public URL findResource(String name) {
     return url != null ? url : directDescendant.get().findResource(name);
   }
 
+  @Override
+  public Enumeration<URL> findResources(String name) throws IOException {
+    final Enumeration<URL> parentResources = super.findResources(name);
+
+    if (directDescendant.get() == null) {
+      return parentResources;
+    }
+
+    final Enumeration<URL> directDescendantResources = directDescendant.get().findResources(name);
+    return new Enumeration<URL>() {
+      @Override
+      public boolean hasMoreElements() {
+        return parentResources.hasMoreElements() || directDescendantResources.hasMoreElements();
+      }
+
+      @Override
+      public URL nextElement() {
+        if (parentResources.hasMoreElements()) {
+          return parentResources.nextElement();
+        }
+        return directDescendantResources.nextElement();
```

Comment:
I am unsure about the "right" order here, as it matters for https://github.com/lightbend/config/blob/c17d58e09af11f9e2a9ddae2eccca18f5f0f01f6/HOCON.md#conventional-configuration-files-for-jvm-apps for example.

Reply:
I’m wondering if the two enumerations should be eagerly merged and stored in memory. Naively it looks possible that if the same resource was found in multiple jars across layers that it would be returned twice in the enumeration?

Reply:
> Naively it looks possible that if the same resource was found in multiple jars across layers that it would be returned twice in the enumeration? Or they could be lazily merged like what is done here but with a filter to exclude resources from the descendant classloader that have the same name as resources loaded by the parent loader.

The resources returned here have the same name by definition, don't they? Or do you mean that we should de-dup based on the full URL, so that a JAR that would be references by both classloaders would only returns its resource once?

Reply:
> I think the ordering makes sense that resources come from the parent first.

Looking at the javadoc, that's indeed what's expected :+1:

https://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html#getResources-java.lang.String-
>  The search order is described in the documentation for `getResource(String)` 

https://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html#getResource-java.lang.String-
> This method will first search the parent class loader for the resource

Reply:
I guess what I am confused on is whether `findResources` should return multiple `URL`s with the same resource name or if like `findResource` it should only return one per resource. Now that I’ve typed that out, it seems to me that the code is probably doing the right thing.